src/panels/lovelace/editor/config-elements/elements/hui-state-badge-element-editor.ts

```diff
@@ -0,0 +1,73 @@
+import { html, LitElement, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { fireEvent } from "../../../../../common/dom/fire_event";
+import type { SchemaUnion } from "../../../../../components/ha-form/types";
+import type { HomeAssistant } from "../../../../../types";
+import "../../../../../components/ha-form/ha-form";
+import { LovelacePictureElementEditor } from "../../../types";
+import { StateBadgeElementConfig } from "../../../elements/types";
+
+const SCHEMA = [
+  { name: "entity", selector: { entity: {} } },
+  { name: "title", selector: { text: {} } },
+  {
+    name: "tap_action",
+    selector: {
+      ui_action: {},
+    },
+  },
+  {
+    name: "hold_action",
+    selector: {
+      ui_action: {},
+    },
+  },
+  { name: "style", selector: { object: {} } },
+] as const;
+
+@customElement("hui-state-badge-element-editor")
+export class HuiStateBadgeElementEditor
+  extends LitElement
+  implements LovelacePictureElementEditor
+{
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @state() private _config?: StateBadgeElementConfig;
+
+  public setConfig(config: StateBadgeElementConfig): void {
+    this._config = config;
```

Comment:
These should probably also check the config with the struct?

Reply:
I removed the struct validation for elements from the card editor, and put it in the element editors themselves. That way individual element validation failure doesn't yank the whole card out of visual mode, but just the single element will get a yaml editor, and everything else is still visually editable. Seems to work as intended, though this looks like a design departure from how entities-card editor handled row validation, which was sort of what I was basing this off of. 

So I'm not sure if that will have any other consequences.